lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -58,11 +58,15 @@ public struct ObservableMacro {
   static func withMutationFunction(_ observableType: TokenSyntax) -> DeclSyntax {
     return 
       """
-      internal nonisolated func withMutation<Member, MutationResult>(
-        keyPath: KeyPath<\(observableType), Member>,
-        _ mutation: () throws -> MutationResult
-      ) rethrows -> MutationResult {
-        try \(raw: registrarVariableName).withMutation(of: self, keyPath: keyPath, mutation)
+      @_disfavoredOverload
```

Comment:
why the disfavor? shouldn't the more specific generic win-out?
also it needs to have an overload to both the `Equatable` version and likely the old method too. The old version probably needs to be there so we ensure no one is broken by the change.

Reply:
Because this function is not generic, so complier will always use this one

Reply:
Although the `Member` is generic, but once I add Equatable to Member in a extension, for example, in the extension of Observable, Complier will only use the version without Equatable

Reply:
Add disfavor can prevent this problem

Reply:
How could this be done in an extension? It has to be contained in the macro emission itself. If the members emitted by the macro include both a method for mutation with Equatable and also a method without Equatable it will solve the problem as you were spelling out in the forums but it won't need to have any disambiguating annotations.

Reply:
It was mentioned in the discussion that support for Equatable doesn't necessarily meet the goal, so I'm going to leave it up to the developers to decide if they want to add support for it. The easiest way to support all types is to add them to an extension. But then I remembered that _$observationRegistrar is private, so unless _$observationRegistrar is internal, it won't not work.

Reply:
Let me remove disfavor for now, developers can still add withMutation manually, or write a macro to do so.